### PR TITLE
feat: implement cross_entropy

### LINKED ISSUE #23462: cross_entropy

### PR DESCRIPTION
**Pull request by HadeerArafa**
# PR Description
implementing cross-entropy loss function for => Extend Ivy Functional API #3856

Close #23462

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by HadeerArafa** [comment]
@MuhammedAshraf2020 can you review my code it passed all the tests on my local repo

**Comment by MuhammedAshraf2020** [comment]
Hello @HadeerArafa! Thank you for your contribution. I appreciate your patience, and I'll review your pull request soon. If you have any questions or need further assistance, feel free to ask.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 1]
```diff
@@ -6,6 +6,74 @@
...
+    n = draw(st.integers(1, 5))
+    dtype_and_input = draw(
+        helpers.dtype_and_values(
+            num_arrays=1,
+            available_dtypes=["float32", "float64"],
+            min_value=-10,
+            max_value=10,
+            allow_inf=False,
+            min_num_dims=2,
+            max_num_dims=2,
```
Comment (MuhammedAshraf2020) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:25]: I believe this doesn't encompass the entire test space, as there's no need to confine the input shape to only two dimensions.

Comment (MuhammedAshraf2020) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:25]: I believe this approach does not cover the entire range of testing scenarios, as there is no requirement to restrict the input shape to only two dimensions; it can be 1, 3, or even more. Therefore, I recommend not setting the minimum and maximum number of dimensions to be 2.
